modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/options.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+function assertOptionParameters<T>({
+	editingLanguageId,
+	multiple,
+	option,
+	valueArray,
+}: {
+	editingLanguageId: Locale;
+	multiple: boolean;
+	option: Option<T>;
+	valueArray: T[];
+}): {
+	active: boolean;
+	checked: boolean;
+	label: string | undefined;
+	type: 'checkbox' | 'item';
+	value: T;
+} {
+	const included = valueArray.includes(option.value);
+
+	const label =
+		typeof option.label === 'string'
+			? option.label
+			: option.label[editingLanguageId];
+
+	return {
+		...option,
+		active: !multiple && included,
+		checked: multiple && included,
+		label,
+		type: multiple ? 'checkbox' : 'item',
+	};
+}
+
+export function normalizeOptions({
+	editingLanguageId,
+	fixedOptions,
+	multiple,
+	options,
+	showEmptyOption,
+	valueArray,
+}: {
+	editingLanguageId: Locale;
+	fixedOptions: Option<string>[];
+	multiple: boolean;
+	options: Option<string>[];
+	showEmptyOption: boolean;
+	valueArray: string[];
+}) {
+	const newOptions: {
+		separator?: boolean;
+		active?: boolean;
+		checked?: boolean;
+		label: string | undefined;
+		type?: 'checkbox' | 'item';
+		value: string | null;
+	}[] = [
+		...options.map((option, index) => ({
+			...assertOptionParameters<string>({
+				editingLanguageId,
+				multiple,
+				option,
+				valueArray,
+			}),
+			separator:
+				Array.isArray(fixedOptions) &&
```

Comment:
We know that `fixedOptions` is an array, otherwise the line 83 `fixedOptions.map()` would rise an exception.

Remove this assertion:
```suggestion
```